client/my-sites/navigation/index.jsx

```diff
@@ -30,7 +39,7 @@ class MySitesNavigation extends React.Component {
 					siteBasePath={ this.props.siteBasePath }
 					onClose={ this.preventPickerDefault }
 				/>
-				<SidebarComponent path={ this.props.path } siteBasePath={ this.props.siteBasePath } />
+				{ asyncSidebar }
```

Comment:
After we move the `config.isEnabled( 'jetpack-cloud' )` logic above to `asyncProps`, this won't need to be a separate variable, and we could just have `<AsyncLoad { ...asyncProps } />` or something along those lines.